Lib/argparse.py

```diff
@@ -2605,9 +2605,9 @@ def print_help(self, file=None):
 
     def _print_message(self, message, file=None):
         if message:
-            if file is None:
-                file = _sys.stderr
-            file.write(message)
+            file = file or _sys.stderr
+            if file is not None:
+                file.write(message)
```

Comment:
Catch issues with unwritable non-None files with
```suggestion
            try:
                file.write(message)
            except (AttributeError, OSError):
                pass
```